src/components/components/login/MetaverseRegister.vue

```diff
@@ -73,33 +73,93 @@
+        label="Advanced"
+        expand-icon-class="text-red"
+        header-class="clean-header"
+    >
+        <q-card class="no-padding">
+            <q-card-section class="no-padding">
+                <q-input
+                    v-model="metaverseServerSetting"
+                    label="Metaverse Server"
+                >
+                    <template v-slot:before>
+                        <q-icon name="language" size="18px" color="primary" />
+                    </template>
+
+                    <template v-slot:append>
+                        <q-icon
+                            name="save"
+                            size="18px"
+                            color="primary"
+                            @click="
+                                metaverseServerStore = metaverseServerSetting;
+                                metaverseServerSettingExpansion = false;
+                            "
+                            class="cursor-pointer"
+                        />
+                    </template>
+                </q-input>
+            </q-card-section>
+        </q-card>
+    </q-expansion-item>
 </template>
 
 <script lang="ts">
 import { defineComponent } from "vue";
 import { useQuasar } from "quasar";
+import { Mutations as StoreMutations } from "@Store/index";
+import { Utility } from "@Modules/utility";
 
 import { Account } from "@Modules/account";
 
 export default defineComponent({
     name: "MetaverseRegister",
 
-    props: ["metaverseUrl"],
+    data: function() {
+        return {
+            username: "",
+            email: "",
+            password: "",
+            confirmPassword: "",
+            showPassword: false,
+            showConfirmPassword: false,
+            metaverseServerSettingExpansion: false,
+            metaverseServerSetting: this.$store.state.metaverse.server,
+            defaultMetaverseServer: this.$store.state.metaverse.server
+        };
+    },
 
-    data: () => ({
-        username: "",
-        email: "",
-        password: "",
-        confirmPassword: "",
-        showPassword: false,
-        showConfirmPassword: false
-    }),
+    computed: {
+        metaverseServerStore: {
+            get() {
+                return this.$store.state.metaverse.server;
+            },
+            async set(newValue: string) {
+                this.$store.commit(StoreMutations.MUTATE, {
+                    property: "metaverse.server",
+                    value: newValue
+                });
+                await Utility.connectionSetup(
+                    this.$store.state.metaverse.server
+                );
+            }
+        }
+    },
 
     methods: {
         async onSubmit() {
             const $q = useQuasar();
             try {
-                const awaiting = await Account.createAccount(this.username, this.password, this.email, this.metaverseUrl);
+                await Utility.connectionSetup(
+                    this.$store.state.metaverse.server
```

Comment:
Why is this call here? If the metaverse URL was changed, there should have already been a call to `metaverseConnectionSetup()`. Is there a case where this has not happened?

Reply:
Fixed